packages/stacks/src/transactions/generate-unsigned-transaction.ts

```diff
@@ -24,60 +24,58 @@ export function initNonce(nonce: number | string) {
   return new BigNumber(nonce, 10);
 }
 
-export function generateUnsignedContractCall(options: StacksUnsignedContractCallOptions) {
-  const { fee, functionArgs, nonce, postConditions } = options;
-
-  const fnArgs = functionArgs.map(arg => deserializeCV(hexToBytes(cleanHex(arg))));
-
-  const parsedOptions = {
-    fee: fee.amount.toString(),
-    functionArgs: fnArgs,
-    nonce: initNonce(nonce).toString(),
-    postConditions,
+export function getUnsignedContractCallParsedOptions(options: StacksUnsignedContractCallOptions) {
+  return {
+    ...options,
+    fee: options.fee.amount.toString(),
+    functionArgs: options.functionArgs.map(arg => deserializeCV(hexToBytes(cleanHex(arg)))),
+    nonce: initNonce(options.nonce).toString(),
+    // `@stacks/transactions` type here is incorrect, it should allow PostConditionWire[]
+    postConditions: options.postConditions,
   };
-
-  return makeUnsignedContractCall({ ...options, ...parsedOptions });
 }
 
-export function generateUnsignedContractDeploy(options: StacksUnsignedContractDeployOptions) {
-  const { fee, nonce, postConditions } = options;
-
-  const parsedOptions = {
-    fee: fee.amount.toString(),
-    nonce: initNonce(nonce).toString(),
-    postConditions: getPostConditions(postConditions?.map(pc => ensurePostConditionWireFormat(pc))),
+export function getUnsignedContractDeployParsedOptions(
+  options: StacksUnsignedContractDeployOptions
+) {
+  return {
+    ...options,
+    fee: options.fee.amount.toString(),
+    nonce: initNonce(options.nonce).toString(),
+    postConditions: getPostConditions(
+      options.postConditions?.map(pc => ensurePostConditionWireFormat(pc))
+    ),
   };
-
-  return makeUnsignedContractDeploy({ ...options, ...parsedOptions });
 }
 
-export function generateUnsignedStxTokenTransfer(options: StacksUnsignedTokenTransferOptions) {
-  const { amount, fee, nonce } = options;
-
-  const parsedOptions = {
-    amount: amount.amount.toString(),
-    fee: fee.amount.toString(),
-    nonce: initNonce(nonce).toString(),
+export function getUnsignedStxTokenTransferParsedOptions(
+  options: StacksUnsignedTokenTransferOptions
+) {
+  return {
+    ...options,
+    amount: options.amount.amount.toString(),
+    fee: options.fee.amount.toString(),
+    nonce: initNonce(options.nonce).toString(),
   };
-
-  return makeUnsignedSTXTokenTransfer({ ...options, ...parsedOptions });
 }
 
-export async function generateStacksUnsignedTransaction(options: StacksUnsignedTransactionOptions) {
-  const { txType } = options;
+export function generateStacksUnsignedTransaction() {
+  return async (options: StacksUnsignedTransactionOptions) => {
```

Comment:
Not sure I fully see the benefit/need of creating a HOF here? There are no outer arguments, so the resulting fn is always the same.

Reply:
Good point, I'll change it today.